tkg/client/upgrade_cluster.go

```diff
@@ -196,9 +196,9 @@ func (c *TkgClient) DoLegacyClusterUpgrade(regionalClusterClient, currentCluster
 	// The addons should upgrade prior to cluster upgrade to account for forward compatibility
 	// i.e. some old addons may not run on the nodes with new k8s version
 	// We will ensure backward compatibility when shipping packages going forward
-	// With package-package-lcm approach addons will be upgraded as part of management package upgrade
+	// With cluster class approach addons will be upgraded as part of management package upgrade
 	// and we do not need to upgrade addons with below function
-	if !options.SkipAddonUpgrade && !config.IsFeatureActivated(constants.FeatureFlagPackageBasedLCM) {
+	if !options.SkipAddonUpgrade && (!config.IsFeatureActivated(constants.FeatureFlagPackageBasedCC) || config.IsFeatureActivated(constants.FeatureFlagAllowLegacyCluster)) {
```

Comment:
Why is this an OR?

Does this means when the FeatureFlagPackageBasedCC and FeatureFlagAllowLegacyCluster are both true, the ytt layer upgrade is called? But when FeatureFlagPackageBasedCC is enabled, the ytt layer should not be called

Reply:
According to the discussion in slack, I have modified this logic. Only use `FeatureFlagPackageBasedCC ` to control if addons need to be upgraded  or not. Pls help review again, thanks!